### ISSUE TITLE
Console logging

### ISSUE THREAD

**Issue by andyjmc** [datakurre/robotkernel#11]
Is it possible to have live console logging? It seems that it only logs everything at the end of the tests?

**Comment by datakurre** [comment]
@andyjmc Could you be more verbose what kind of logging you are after.

Currently it show “additive progress dots” during execution and only shows logs on error. On success, log is only available as link.

I’d like to understand, what extra information the live console log would provide? (Jupyter UI encourages to split scripts into small cells [user experience is not good with huge cells] - like single test per cell - and that makes console log less interesting).

**Comment by andyjmc** [comment]
If you run pybot from bash you can use`from robot.api import logger` and the `logger.console('Insert string here...')` you can print messages while the tests are being run. This is very useful of you are downloading or testing large datasets as it can inform the user about how many items have been tested etc... 
See example screenshot below...

![screenshot 2018-12-11 at 17 51 27](https://user-images.githubusercontent.com/42138224/49819540-a2d60e00-fd6d-11e8-9443-5d4049fcf145.png)

**Comment by datakurre** [comment]
@andyjmc I saw you mentioning that your are trying robot kernel in multikernel notebook. If you have time, I’d love to read briefly how have you been using it and how well it has worked. I have very little experience on multikernel notebooks.

**Comment by andyjmc** [comment]
> Would it be ok to only show logger.console messages without those usual surroundings? And would it also be ok to remove the messages after execution (similarly to the progress dots).


I think it would be fine to remove the messages after execution. It would also be fine to show the messages without the usual surroundings but it would be useful to know when a new test has begun/finished

>  I saw you mentioning that your are trying robot kernel in multikernel notebook. If you have time, I’d love to read briefly how have you been using it and how well it has worked. I have very little experience on multikernel notebooks.

There is a Kernal called SoS which can be used to host multiple kernels at once. I used this to get your RF kernel with a python3 kernel on the same Jupyter page - https://vatlab.github.io/sos-docs/

**Comment by datakurre** [comment]
Thanks again. Possibly I could show: progress dots, the current test, the current keyword, cumulatice console log messages.

Could you tell more about your use case for SoS? Why not separate Python and RF notebooks? Did it work out of the box?

**Comment by datakurre** [comment]
FYI there is support for a single magic for defining a keyword library within RF notebook

https://github.com/datakurre/robotkernel/blob/master/examples/OpenCV.ipynb

But sure that is missing all ipykernel features and also syntax highlighting. (But it is supported by “nbrobot” script provided by the latest release.)

For importing a separate python notebook as a keyword library I added support for nbimporter (when available). Unfortunately, with that, it seems that keyword library from a python notebook gets reloaded only with kernel restart.

**Comment by andyjmc** [comment]
> Thanks again. Possibly I could show: progress dots, the current test, the current keyword, cumulatice console log messages.
That would be awesome. Thank you! 
> Could you tell more about your use case for SoS? Why not separate Python and RF notebooks? Did it work out of the box?

We need to run custom (written by our engineers) python in the same notebook along side the standard RF tests which we write. Because of our structure it makes is much easier to run it all in the same notebook without having to jump around. It didn't completely work out of box but the developer of SoS was kind enough to make some changes so now it should.

> FYI there is support for a single magic for defining a keyword library within RF notebook
>https://github.com/datakurre/robotkernel/blob/master/examples/OpenCV.ipynb

That is awesome. Thanks!

**Comment by datakurre** [comment]
Cumulative logging made the user experience too flickery for my taste, so my implementation ended up to be:
```
(throbber) | Test name | Keyword name | Last console log message
```

**Comment by datakurre** [comment]
Released in 0.9.0.
